lib/client/native_client.dart

```diff
@@ -0,0 +1,178 @@
+import 'dart:ffi';
+import 'dart:io' show Platform;
+
+import 'package:ffi/ffi.dart';
+
+typedef InitDartApiNative = IntPtr Function(Pointer<Void>);
+typedef InitDartApi = int Function(Pointer<Void>);
+
+typedef NewClientNative = IntPtr Function(
+    Pointer<Utf8> appId,
+    Pointer<Utf8> rendezvousUrl,
+    Pointer<Utf8> transitRelayUrl,
+    Int32 passPhraseComponentLength);
+typedef NewClient = int Function(
+    Pointer<Utf8> appId,
+    Pointer<Utf8> rendezvousUrl,
+    Pointer<Utf8> transitRelayUrl,
+    int passPhraseComponentLength);
+
+typedef ClientSendTextNative = Int32 Function(Uint32 goClientId,
+    Pointer<Utf8> msg, Pointer<Pointer<Utf8>> codeOut, Int32 callbackPortId);
+
+typedef ClientSendText = int Function(int goClientId, Pointer<Utf8> msg,
+    Pointer<Pointer<Utf8>> codeOut, int callbackPortId);
+
+typedef ClientSendFileNative = Int32 Function(
+    Uint32 goClientId,
+    Pointer<Utf8> fileName,
+    Uint32 length,
+    Pointer<Uint8> fileBytes,
+    Pointer<Pointer<Utf8>> codeOut,
+    Int32 callbackPortId);
+
+typedef ClientSendFile = int Function(
+    int goClientId,
+    Pointer<Utf8> fileName,
+    int length,
+    Pointer<Uint8> fileBytes,
+    Pointer<Pointer<Utf8>> codePtr,
+    int callbackPortId);
+
+typedef ClientRecvTextNative = Int32 Function(
+    Uint32 goClientId, Pointer<Utf8> code, Int32 callbackPortId);
+
+typedef ClientRecvText = int Function(
+    int goClientId, Pointer<Utf8> code, int callbackPortId);
+
+typedef ClientRecvFileNative = Int32 Function(
+    Uint32 goClientId, Pointer<Utf8> code, Int32 callbackPortId);
+
+typedef ClientRecvFile = int Function(
+    int goClientId, Pointer<Utf8> code, int callbackPortId);
+
+const DEFAULT_APP_ID = "lothar.com/wormhole/text-or-file-xfer";
```

Comment:
Allow override via environment variables or config file at build-time, rely on go for defaults.